src/commands/sensor/aggregates/sensor.aggregate.ts

```diff
@@ -0,0 +1,153 @@
+    active: boolean, observationArea: object, documentationUrl: string, category: string, theme: string,
+    typeName: string, typeDetails: object) {
+    this.apply(new SensorCreated(this.aggregateId, nodeId, ownerIds, name, location, 
+      aim, description, manufacturer, active, observationArea, documentationUrl, 
+      category, theme, typeName, typeDetails));
+
+    for (const dataStream of dataStreams) {
+      this.apply(new DataStreamCreated(this.aggregateId, dataStream.id, dataStream.name, dataStream.description,
+        dataStream.unitOfMeasurement, dataStream.isPublic, dataStream.isOpenData, dataStream.isReusable,
+        dataStream.documentationUrl, dataStream.dataLink, dataStream.dataFrequency, dataStream.dataQuality));
+    }
+  }
+
+  createDataStream(dataStreamId: string, name: string, description: string, unitOfMeasurement: string,
+    isPublic: boolean, isOpenData: boolean, isReusable: boolean, documentationUrl: string,
+    dataLink: string, dataFrequency: number, dataQuality: number) {
+    this.apply(new DataStreamCreated(this.aggregateId, dataStreamId, name, description, unitOfMeasurement, 
+      isPublic, isOpenData, isReusable, documentationUrl, dataLink, dataFrequency, dataQuality));
+  }
+
+  deleteDataStream(sensorId: string, dataStreamId: string) {
+    this.apply(new DataStreamDeleted(this.aggregateId, sensorId, dataStreamId));
+  }
+
+  update(name: string, aim: string, description: string, manufacturer: string, 
+    observationArea: object, documentationUrl: string, category: string, theme: string,
+    typeName: string, typeDetails: object) {
+    this.apply(new SensorUpdated(this.aggregateId, name, aim, description, manufacturer, 
+      observationArea, documentationUrl, category, theme, typeName, typeDetails));
+  }
+
+  transferOwnership(oldOwnerId: string, newOwnerId: string) {
+    this.apply(new SensorOwnershipTransferred(this.aggregateId, oldOwnerId, newOwnerId));
+  }
+
+  shareOwnership(ownerId: string) {
+    this.apply(new SensorOwnershipShared(this.aggregateId, ownerId));
+  }
+
+  updateLocation(lat: number, lon: number, height: number, baseObjectId: string) {
+    this.apply(new SensorLocationUpdated(this.aggregateId, lat, lon, height, baseObjectId));
+  }
+
+  activate() {
+    if (this.state.active) throw new SensorActiveException(this.state.id);
+
+    this.apply(new SensorActivated(this.aggregateId));
+  }
+
+  deactivate() {
+    if (!this.state.active) throw new SensorInActiveException(this.state.id);
+
+    this.apply(new SensorDeactivated(this.aggregateId));
+  }
+
+  delete() {
+    this.apply(new SensorDeleted(this.aggregateId));
+  }
+
+  private onSensorCreated(event: SensorCreated) {
+    this.state = new SensorStateImpl(this.aggregateId);
+
+    this.state.actives.push(event.data["active"]);
+  }
+
+  private onDataStreamCreated(event: DataStreamCreated) {
+    // Called on DataStreamCreated -> Update Database.
```

Comment:
Dit comments is niet waar, toch? Database wordt bijgewerkt vanuit ... euh, hoe heet die hier? ik noem 'm maar even `EventProcessor`. Deze listeners zijn bedoeld om de in memory state bij te werken (en dat doe je hieronder ook in de andere methods (bijv. in `onActivated`), right?

Reply:
Klopt, is nog een oud comment :)